### PR TITLE
Refactor server endpoint management, fix shutdown

### LINKED ISSUE #209: When I ctl-C kill the spire server, I get an error
When I invoke SPIRE-server from a terminal, eg. `$ spire-server run` and then ctl-C to kill the process, I get an error as part of the shutdown sequence. eg.

```
^X^CINFO[0002] Stopping plugins...                          
2017/09/21 23:06:23 [DEBUG] plugin: waiting for all plugin processes to complete...
ERRO[0003] interrupt  
```

While the behavior is exactly what I would expect, I'm surprised that the interrupt is being reported as an error, since this seems like a legitimate way to end the process. Am I doing it wrong? Or is this not really an error?

(Ubuntu 14.04)

### PR DESCRIPTION
**Pull request by evan2645**
This commit refactors the server to move endpoint lifecycle management to a dedicated package. This was done in reaction to #209, as properly addressing the clean shutdown problem in the server was difficult to do without it.

As such, this fixes #209

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,8 +12,8 @@ script:
     - ./build.sh vendor
     - ./build.sh utils
     - ./build.sh protobuf_verify
-    - ./build.sh binaries
     - ./build.sh test
+    - ./build.sh binaries
```
Comment (evan2645) [inline .travis.yml]: Brrr I meant to send this in a separate PR. @drrt this is just something I noticed while watching the travis builds, LMK if it doesn't make sense and/or if I should back it out

**Comment by ajessup** [comment]
Yaaaasss!
